core/client/fs/src/main/java/alluxio/client/block/stream/GrpcDataReader.java

```diff
@@ -113,6 +113,11 @@ public DataBuffer readChunk() throws IOException {
       if (message != null) {
         response = message.getMessage();
         buffer = message.getBuffer();
+        if (buffer == null && response.hasChunk() && response.getChunk().hasData()) {
+          // falls back to use chunk message for compatibility
+          ByteBuffer byteBuffer = response.getChunk().getData().asReadOnlyByteBuffer();
```

Comment:
Is there ever a case where `message.getMessage()` could return `null` for response?

Reply:
`message.getMessage()` never returns null because it is only created when gRPC returns a valid non-null response.